kp_test.go

```diff
@@ -1509,3 +1508,37 @@ func TestRestoreKey(t *testing.T) {
 
 	assert.True(t, gock.IsDone(), "Expected HTTP requests not called!")
 }
+
+func TestInitiate_DualAuthDelete(t *testing.T) {
+	defer gock.Off()
+
+	gock.New("http://example.com").
+		MatchParam("action", "setKeyForDeletion").
+		Reply(204)
+
+	c, _, err := NewTestClient(t, nil)
+	gock.InterceptClient(&c.HttpClient)
+	defer gock.RestoreClient(&c.HttpClient)
+	c.tokenSource = &FakeTokenSource{}
+
+	err = c.InitiateDualAuthDelete(context.Background(), "keyID")
+
+	assert.Nil(t, err)
```

Comment:
You need a check here to make sure we actually intercepted and responded to the request properly using gock.

This is the line that does that from other tests
```
	assert.True(t, gock.IsDone(), "Expected HTTP requests not called!")
```